src/studio/src/designer/frontend/packages/schema-editor/src/components/SchemaInspector.tsx

```diff
@@ -294,8 +310,11 @@ const SchemaInspector = ((props: ISchemaInspectorProps) => {
     />;
   });
 
+  const onRestrictionReturn = (e: any) => {
+    onAddRestrictionClick(e);
+  };
   const renderItemRestrictions = (item: UiSchemaItem) => item.restrictions?.map((field: Field) => {
-    if (!field || field.key.startsWith('@')) {
+    if (!field.key && field.key.startsWith('@')) {
```

Comment:
I'm not sure what a 'key' is in this context, but will this ever be true? Shouldn't the first '!' be removed?

Reply:
This formally returned null if there was no key, that means the key in RestictionField, can be 'null' or 'undefined' and since this is a map function, that can quickly lead to more than one key with the same value...

Reply:
Right, fixed